src/pipelinedefinition.cpp

```diff
@@ -216,92 +216,300 @@ void PipelineDefinition::makeSubscriptions(ModelManager& manager) {
-        if (!result.ok()) {
-            SPDLOG_ERROR("Validation of pipeline({}) definition failed. Missing model: {} version: {}", this->pipelineName, node.modelName, node.modelVersion.value_or(0));
-            return StatusCode::MODEL_NAME_MISSING;
+class NodeValidator {
+    const std::string& pipelineName;
+    ModelManager& manager;
+    const NodeInfo& dependantNodeInfo;
+    const pipeline_connections_t& connections;
+    const std::vector<NodeInfo>& nodeInfos;
+
+    std::unique_ptr<ModelInstanceUnloadGuard> dependantModelUnloadGuard;
+    std::shared_ptr<ModelInstance> dependantModelInstance;
+    std::set<std::string> remainingUnconnectedDependantModelInputs;
+
+public:
+    NodeValidator(
+        const std::string& pipelineName,
+        ModelManager& manager,
+        const NodeInfo& dependantNodeInfo,
+        const pipeline_connections_t& connections,
+        const std::vector<NodeInfo>& nodeInfos) :
+        pipelineName(pipelineName),
+        manager(manager),
+        dependantNodeInfo(dependantNodeInfo),
+        connections(connections),
+        nodeInfos(nodeInfos) {
+        SPDLOG_DEBUG("Validation of pipeline: {}; node name: {}; node kind: {}",
+            pipelineName,
+            dependantNodeInfo.nodeName,
+            dependantNodeInfo.kind);
+    }
+
+    Status fetchUnderlyingModelInstance() {
+        if (!getModelInstance(
+                manager,
+                dependantNodeInfo.modelName,
+                dependantNodeInfo.modelVersion.value_or(0),
+                dependantModelInstance,
+                dependantModelUnloadGuard)
+                 .ok()) {
+            SPDLOG_ERROR("Validation of pipeline({}) definition failed. Missing model: {}; version: {}",
+                pipelineName,
+                dependantNodeInfo.modelName,
+                dependantNodeInfo.modelVersion.value_or(0));
+            return StatusCode::PIPELINE_NODE_REFERING_TO_MISSING_MODEL;
         }
+        return StatusCode::OK;
+    }
+
+    Status getDependencyNodeInfo(const std::string& dependencyNodeName, std::vector<NodeInfo>::const_iterator& dependencyNodeInfo) {
+        // Find dependency node info object.
+        dependencyNodeInfo = std::find_if(
+            std::begin(this->nodeInfos),
+            std::end(this->nodeInfos),
+            [dependencyNodeName](const NodeInfo& nodeInfo) { return nodeInfo.nodeName == dependencyNodeName; });
+        if (dependencyNodeInfo == std::end(this->nodeInfos)) {
+            SPDLOG_ERROR("Validation of pipeline({}) definition failed. Node (name:{}) is connected to missing dependency node (name:{})",
+                pipelineName,
+                dependantNodeInfo.nodeName,
+                dependencyNodeName);
+            return StatusCode::PIPELINE_NODE_REFERING_TO_MISSING_NODE;
+        }
+
+        // Exit cannot be dependency of any node.
+        if (dependencyNodeInfo->kind == NodeKind::EXIT) {
+            return StatusCode::UNKNOWN_ERROR;
```

Comment:
Don't we have any more telling status code? I am not sure if this would fall out at the stage of json schema validation.

Reply:
I was sure that we cannot reach this part of code due to earlier stages of validation. Just checked and this is possible. Adding new status code and unit test for this scenario.

Reply:
Done